src/libexpr-c/nix_api_expr.h

```diff
@@ -14,6 +14,16 @@
 #include "nix_api_util.h"
 #include <stddef.h>
 
+#ifndef __has_c_attribute
+#define __has_c_attribute(x) 0
+#endif
+
+#if __has_c_attribute(deprecated)
+#  define DEPRECATED(msg) [[deprecated(msg)]]
```

Comment:
Since this is a public header, we should probably namespace this macro to avoid conflicts with other headers that might use a similar trick.